### PR TITLE
Reword repository indicator fetch setting

### PR DESCRIPTION
**Pull request by tidy-dev**
Related Issues:
https://github.com/desktop/desktop/issues/10687
[GitHub Desktop fetching even with auto fetch disabled ](https://github.com/github/desktop/issues/794)

## Description
The "Periodically fetch and refresh status of all repositories" is to toggle the background fetching of repositories in order to display the status indicators in the repository list.  Unfortunately, we have had users interpret the setting to mean it should  stop all the periodic fetching including their currently open one. However, this is not true, GitHub Desktop will still periodically fetch the **open** repository in order to keep it in sync with it's remote.

Further explanation of why this doesn't turn off fetching in the currently open repository:
> As part of these background fetches we fast forward all but the currently selected branch which in turn powers "Update from default branch" and the merge/rebase dialogs as they operate on local branches only. It's also what enables us to show the ahead/behind indicator in the push/pull toolbar (and what enables us to detect that we need to do a pull vs a fetch).

Therefore, having a setting to turn all periodic fetching off would require a bit of research to determine all the consequences and handle all the use cases of the open repository.

### Screenshots

image

## Release notes

Notes: [Improved] Clarified the outcome of toggling the setting under "Background Updates" in the "Advanced" settings.

### CODE REVIEW

File: app/src/ui/preferences/advanced.tsx [hunk 1]
```diff
@@ -76,17 +76,22 @@ export class Advanced extends React.Component<
...
                 : CheckboxValue.Off
             }
             onChange={this.onRepositoryIndicatorsEnabledChanged}
+            ariaDescribedBy="periodic-fecth-desription"
           />
-          <p className="git-settings-description">
-            Allows the display of up-to-date status indicators in the repository
-            list. Disabling this may improve performance with many repositories.
+          <p
+            id="periodic-fecth-desription"
```
Comment (niik) [inline app/src/ui/preferences/advanced.tsx]:
```suggestion
            id="periodic-fecth-description"
```

File: app/src/ui/preferences/advanced.tsx [hunk 2]
```diff
@@ -76,17 +76,22 @@ export class Advanced extends React.Component<
...
           <Checkbox
-            label="Periodically fetch and refresh status of all repositories"
+            label="Periodically refresh the status indicators in repository list"
             value={
               this.props.repositoryIndicatorsEnabled
                 ? CheckboxValue.On
                 : CheckboxValue.Off
             }
             onChange={this.onRepositoryIndicatorsEnabledChanged}
+            ariaDescribedBy="periodic-fecth-desription"
```
Comment (niik) [inline app/src/ui/preferences/advanced.tsx]:
```suggestion
            ariaDescribedBy="periodic-fecth-description"
```

File: app/src/ui/preferences/advanced.tsx [hunk 3]
```diff
@@ -76,17 +76,22 @@ export class Advanced extends React.Component<
...
-          <p className="git-settings-description">
-            Allows the display of up-to-date status indicators in the repository
-            list. Disabling this may improve performance with many repositories.
+          <p
+            id="periodic-fecth-desription"
+            className="git-settings-description"
+          >
+            This requires the periodic fetching of repositories that are not
+            currently open. Turning this off will not stop the periodic fetching
+            of your open repository.
```
Comment (niik) [inline app/src/ui/preferences/advanced.tsx]: Open repository is not a terminology we've used in the past. We haven't had to communicate this concept to the user frequently but our internal code comments all talk about the "currently selected repository" and that's how we refer to it in docs.github.com as well.

I don't know what that would do to the sentence above though 🤔 

```suggestion
            of your currently selected repository.
```

**Review by niik — COMMENTED** [review]
This is already loads better than what we've got and I'd be happy to take this improvement. I though about trying an angle where we title the setting after what it enables and not what it does but I couldn't get it to not be too wordy

```
 [ ] Show which repositories have local or remote changes in the repository list
```

**Comment by oscarllann** [comment]
<p className="git-settings-description">
-            Allows the display of up-to-date status indicators in the repository
-            list. Disabling this may improve performance with many repositories.
+          <p
+            id="periodic-fecth-desription"
+            className="git-settings-description"
+          >
+            This requires the periodic fetching of repositories that are not
+            currently open. Turning this off will not stop the periodic fetching
+            of your open repository.

File: app/src/ui/preferences/advanced.tsx (hunk 3 above)
Comment (sergiou87) [inline app/src/ui/preferences/advanced.tsx]: I like @oscarllann angle on performance. Maybe it gets excessively wordy but adding something like... "but will improve overall performance of the app for users with many repositories" could be useful to our users.

**Comment by tidy-dev** [comment]
>  [ ] Show which repositories have local or remote changes in the repository list

I do prefer this approach as it is more direct. I had something long on one iterations but was still using "status indicators", I like the explanation of the behavior better. Probably the only thing I prefer about "status indicators" is that it tells me that is likely referring to the "icon" stuff in the repository list.  I think that this is going to be a verbose setting no matter what because it is hard to concisely communicate. I am going to mull on this a bit see if I can think of anything different.

> may improve performance with many repositories. 

Good call out -> having the reason for the setting makes sense.

> up-to-date status indicators

I was leery of the "up-to-date" verbiage since it is a periodic fetch so it is not always up-to-date :D

**Comment by tidy-dev** [comment]
Latest thoughts.. but the verboseness isn't pretty with the checkbox centering.. But could fix if we decide we like some wording that is two lines.

image

**Comment by sergiou87** [comment]
Yeah that doesn't look good 😭 What do you think about moving some of that info to the description text?

```
[ ] Show status icons in the repository list

These icons will indicate whether repositories have local or remote changes. This requires the periodic fetching of…
```

**Comment by tidy-dev** [comment]
Yeah. I definitely like a one line setting better. But, man, that wall of text. :P I split it into two paragraphs in hopes of making it scannable.

Using one line setting with long explanation split into two paragraphs.

**Comment by tidy-dev** [comment]
Alternative to wall of text above .. more scannable wall of text?

Description broken into sections and bullets

**Comment by sergiou87** [comment]
I prefer the two paragraphs better!

**Comment by tidy-dev** [comment]
PR currently

image

**Review by sergiou87 — APPROVED** [review]
Love it! Thank you 😄
